crates/api/src/lib.rs

```diff
@@ -172,15 +172,6 @@ pub(crate) async fn collect_moderated_communities(
   }
 }
 
-pub(crate) fn check_optional_url(item: &Option<Option<String>>) -> Result<(), LemmyError> {
-  if let Some(Some(item)) = &item {
-    if Url::parse(item).is_err() {
-      return Err(ApiError::err("invalid_url").into());
```

Comment:
This doesn't seem to be in your changes, wherever this parsing has an error, it should return this API error.

Reply:
When the URL is moved into the api models directly, serde deserialization automatically provides an error message. I simulated this by replaying some edited messages on commands which used to go through this validator, and it seems to work, just with a different error message.

- WS Request: `{"op":"CreatePost","data":{"community_id":2,"name":"test 2","nsfw":false,"auth":"...","url":"some invalid url"}}`
- WS Response: `invalid value: string "invalid url!", expected relative URL without a base at line 1 column 200`

Reply:
It should return the exact same error message as before, because that is parsed by lemmy/ui and other clients, and used to show the actual, user friendly error.

Reply:
The problem with that is that we need to return proper `ApiErrors` for the front end clients, they can't read random strings thrown at them. That `invalid_url` string is an `i18n` key that's translated into many languages.

Reply:
The previous API workflows that went through this old `check_optional_url` validation are:
- create community (icon and banner)
- edit community (icon and banner)
- create post (data url)
- save user (avatar and banner)

I believe none of these workflows can be performed with invalid URLs on lemmy-ui. The image URLs all go through special image UI, and the post URL is validated before allowing users to submit. I was only able to trigger those errors by manually running malformed websocket payloads.